### DISCUSSION TITLE
Is it intended behavior for cargo-make not to use environment variable defined in the executing environment?

### DISCUSSION THREAD

**Discussion by wmmc88** [sagiegurari/cargo-make#1169] [Q&A] [answered]
You can define an environment variable for `RUSTFLAGS` via `$env:RUSTFLAGS="-D warnings"` in pwsh or `export RUSTFLAGS="-D warnings"` in bash. Any `cargo build` commands you run after it will use that value for rust flags. 

When you do `cargo make build`, the environment which `cargo-make` invokes `cargo build` does not inherit the environment in which `cargo make` was executed. Is this intentional behavior? I find its sort of weird that it creates this completely clean environment instead of using the original execution environment as a base.

**Comment by sagiegurari** [comment]
its possible that rustflags is defined in the internal makefile, maybe stable.toml and therefore overwirtes external value.
if so, it can be fixed by overriding it in custom make file by setting it to use the already defined env value

**Comment by sagiegurari** [comment]
i see i dknt overwrite it.... i wonder if its same cargo issue you fixed a while back

**Reply by wmmc88** [reply, marked as answer]
Thanks for reminding me about https://github.com/sagiegurari/cargo-make/issues/1059#issuecomment-2021725736. I reread the threads, and it looks like the[ original recommendation ](https://github.com/rust-lang/rustup/issues/3029#issuecomment-1182517215) from `rustup` was to clear `CARGO`, `RUSTC`, and `RUSTDOC`, but in the [original fix](https://github.com/wmmc88/cargo-make/commit/32fc95bccd7e029bb1b36bd923f8a774806b3718), you added code to reset `RUSTC`, `RUSTDOC` and **`RUSTFLAGS`**. I think that to fix this, `RUSTFLAGS` needs to be removed from the list of vars to clear (as it shouldn't have been there in the first place).

**Reply by sagiegurari** [reply]
cool. I'll fix it

**Comment by sagiegurari** [comment]
i'm releasing a fix for this now. will be released in an hour or less.
